synapse/pipeline.yml

```diff
@@ -570,3 +570,46 @@ steps:
           # the hosts Docker.
           volumes:
             - "/var/run/docker.sock:/var/run/docker.sock"
+
+  - command:
+      # Build a docker image from the checked out Synapse source
+      - "docker build -t matrixdotorg/synapse:latest -f docker/Dockerfile ."
+      # Build a second docker image on top of the above image. This one sets up Synapse workers
+      # as well as everything else it needs to run a federating setup
+      - "docker build -t matrixdotorg/synapse:workers -f docker/Dockerfile-workers ."
+      # We use the complement:latest image to provide Complement's dependencies, but want
+      # to actually run against the master branch of Complement, so download it here.
+      - "wget https://github.com/matrix-org/complement/archive/master.tar.gz"
+      - "tar -xzf master.tar.gz"
+      # Build an image for running the above worker setup in Complement. This involves
+      # disabling rate-limiting, using Complement's CA etc.
+      - "docker build -t complement-synapse -f complement-master/dockerfiles/SynapseWorkers.Dockerfile complement-master/dockerfiles"
+      # Finally, compile and run the tests.
+      - "cd complement-master"
+      - "COMPLEMENT_BASE_IMAGE=complement-synapse:latest go test -v -tags synapse_blacklist ./tests"
+    label: "\U0001F9EA Complement | Synapse Workers"
+    agents:
+      # Running every worker takes a beefy system
+      queue: "xlarge"
+    plugins:
+      - docker#v3.7.0:
+          # The dockerfile for this image is at https://github.com/matrix-org/complement/blob/master/dockerfiles/ComplementCIBuildkite.Dockerfile.
+          image: "matrixdotorg/complement:latest"
+          mount-buildkite-agent: false
+          environment:
+            # Complement needs to know if it is running under CI
+            - "CI=true"
+            # Enable debug logging
+            - "COMPLEMENT_DEBUG=1"
```

Comment:
is there a plan to turn this off again, or?

Reply:
Hm, well this parameter will log every request sent to the homeserver with its contents, which I'd find is useful for debugging. It does also print debug logs from Complement, which at the moment are not very much, but may become noisy in the future.

Do we want all client requests to the homeserver logged in CI? Or is that too noisy, and people can re-run the tests locally/check the test code if they'd like to see the exact requests being made?

There's also a separate discussion about potentially splitting this into two environment variables. One for printing client requests, and the other for Complement debug logs.

Reply:
it sounds rather noisy to me. We don't do that for sytest.

Reply:
(alternatively, if it's useful in CI, why wouldn't it be useful always?)

Reply:
Yeah, I think the intention is that it's useful for Complement's CI, but not Synapse's. If we need request logs on failing tests, then we can advocate for that to work outside of the `COMPLEMENT_DEBUG` env var.

I'll remove it.